### PR TITLE
Suggestion feature

### PR DESCRIPTION
**Pull request by nithinmuthukumar**
Adds suggestion feature, that takes in state so that it can in the future apply more complex behaviour

### CODE REVIEW

**Comment by nithinmuthukumar** [comment]
Adds a suggestion feature with example, rough implementation. Uses history to suggest by utilising the search method. This couples suggestions with history which seems to be necessary. 

- The suggestion function could be interchangeable for History such as `DefaultHistory` takes in a search function that sets it up. 

- History could also take in LineCtx, so more complex forms of history such as directory sensitive history could be implemented. This would require the line and core merge.

- Removed `HistoryItem` and replaced with `String`. This was defaulted to String in line anyway and it isn't feasible to have it be dynamic since History is so opaque to line. The inner structure of History can be whatever now however the api deals with `String`

File: crates/shrs_core/src/readline/suggester.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+pub struct DefaultSuggester;
+impl Suggester for DefaultSuggester {
+    fn suggest(&self, line_ctx: &LineStateBundle) -> Option<String> {
+        let h = &line_ctx.ctx.history;
+        let res = line_ctx.line.get_full_command();
+        if res.is_empty() {
+            return None;
+        }
+
+        for i in 0..h.len() {
```
Comment (MrPicklePinosaur) [inline crates/shrs_core/src/readline/suggester.rs:15]: we should really get a proper history iterator soon...

File: crates/shrs_core/src/theme.rs [hunk 2]
```diff
@@ -16,6 +17,7 @@ impl Default for Theme {
             err_style: ContentStyle::new().red(),
             selection_style: ContentStyle::new().white(),
             completion_style: ContentStyle::new().red(),
+            suggestion_style: ContentStyle::new().grey(),
```
Comment (MrPicklePinosaur) [inline crates/shrs_core/src/theme.rs:20]: for me at least the `white` and `grey` colors were identical, not sure how many people will actually have distinct colors for these - maybe should switch to a color that stands out more more suggestion.

**Review by MrPicklePinosaur — APPROVED** [review]
feature works great!

let's make the suggestion also take the current directory into account after we have implemented #422

File: crates/shrs_core/src/readline/suggester.rs (hunk 1 above)
Comment (nithinmuthukumar) [reply crates/shrs_core/src/readline/suggester.rs:15]: Added a history iterator

File: crates/shrs_core/src/theme.rs (hunk 2 above)
Comment (nithinmuthukumar) [reply crates/shrs_core/src/theme.rs:20]: I think it should be unobtrusive, as its a suggestion

**Pull request merged**